### PR TITLE
ON-45175 # add mui theme to calendar booking pages

### PR DESCRIPTION
**Pull request by divporter**
## Requester Checklist

Please only check the items that you have actioned. Do not check items that are not applicable to your PR.

### Implementation

- [ ] Have you tested your implementation locally?
- [ ] If applicable, has an appropriate changelog entry been added? Do not include if you are fixing/changing something that is only in the current release.
- [ ] There are no warnings that have been suppressed unnecessarily
- [ ] Have automated tests been added, or have related ones been updated to cover the change?
- [ ] Have all OneBlink dependency updates been completed (eg apps / apps-react / types etc).
- [ ] Have you ensured this change does not add unwanted dependencies?
- [ ] If this PR contains a refactor, have relevant Jira testing tasks added?
- [ ] Changes that will knowingly make the feature/bug incomplete have been commented with TODO and a description of what needs to be done to finish the feature/bug
- [ ] Any changes made to public APIs have been reflected in the documentation
- [ ] Have you isolated business logic where possible to allow for unit testing?

### Logging and Debugging

- [ ] Are the error messages, if any, informative?
- [ ] Are there enough log events and are they written in a way that allows for easy debugging?
- [ ] "Debugging" code removed
- [ ] Front-end: No erroneous `Console.WriteLines`

### Readability

- [ ] All class, variable, property and method modifiers are provided with the smallest scope possible
- [ ] New files, variables and functions are descriptive/comprehensible and named consistently.
- [ ] There is no dead code (unreachable code)
- [ ] There is no usage of [magic numbers](<https://en.wikipedia.org/wiki/Magic_number_(programming)>)
- [ ] There is no commented out code.
- [ ] In hard-to-understand areas, comments exist and describe rationale or reasons for decisions in code

### Security

- [ ] All personal data inputs are checked (for the correct type, length/size, format, and range).
- [ ] No sensitive information is logged or visible in a stacktrace
- [ ] Are authorization and authentication handled correctly?
- [ ] Is (user) input validated, sanitized, and escaped to prevent security attacks such as cross-site scripting or SQL injection?
- [ ] Is data retrieved from external APIs or libraries checked for security issues?
- [ ] Do API endpoints return appropriate status codes

## Reviewer Guide

- It is important that you understand the purpose of the PR.
- You are encouraged to engage with the requestor if you do not understand any of the proposed code changes/additions/deletions.
- Do you, the reviewer, understand what the code does? Do you think a specific expert, like a security expert or a usability expert, should look over the code before it can be accepted?
- Is a framework, API, library, or service used that should not be used? Are there alternatives you could recommend?
- Are there existing hooks/components/functions in the same code base that could be utilised?
- When reviewing tests, attempt to identify missing edge cases that may be relevant to the proposed implementation
- Ensure you check for:
  - Security
  - Scalability
  - Performance
  - Maintainability

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -263,16 +268,19 @@ export function renderCalendarBookingRescheduleForm(options?: {
...
+        </ModalContainerProvider>
+      </ThemeProvider>
     </React.StrictMode>,
     document.querySelector(selector),
   )
 }
 
 export function renderCalendarBookingCancelForm(options?: {
   selector: string
+  formsAppId: number
```
Comment (mymattcarroll) [inline src/index.tsx:283]: Would be nice to validate this so that devs embedding forms get nice errors if they dont pass this in.

File: src/ThemeProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      },
+      info: {
+        main: '#2d7ab9',
+      },
+      error: {
+        main: '#E8113C',
+      },
+      primary: {
+        main:
+          (state.status === 'SUCCESS' && state.result.styles.highlightColour) ||
```
Comment (mymattcarroll) [inline src/ThemeProvider.tsx:48]:
```suggestion
           state.result.styles.highlightColour ||
```

Dont need to check `status` again, typescript will already know that it is `SUCCESS`

File: src/ThemeProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+          '#407E8C',
+      },
+      secondary: {
+        main:
+          (state.status === 'SUCCESS' &&
```
Comment (mymattcarroll) [inline src/ThemeProvider.tsx:53]: same here.

File: src/ThemeProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        <h3 className="title is-3">Error Loading Configuration</h3>
+        <p className="content has-text-danger">{state.error.message}</p>
+        <button className="button" onClick={refresh}>
+          Try Again
+        </button>
+      </div>
+    )
+  }
+
+  const muiTheme = createTheme({
```
Comment (Zaxist) [inline src/ThemeProvider.tsx:34]: I think it would be beneficial to `memo` this, because it will re-create when `children` changes.

Comment (Zaxist) [reply src/ThemeProvider.tsx:34]: Although, that may never happen.

File: src/ThemeProvider.tsx (hunk 2 above)
Comment (divporter) [reply src/ThemeProvider.tsx:48]: Ah yeah, was before I put the error/loading handling on. what a n00b
